File: src/org/ldccc/om3/dto/DTO.java
package org.ldccc.om3.dto;

import java.util.HashMap;

public class DTO {
	public static final String ID = "ID";

	public HashMap<String, Object> map = new HashMap<>();

	public DTO(HashMap<String, Object> map) {
		this.map = map;
	}

	public DTO() {
	}

	@Override
	public String toString() {
		return map.toString();
	}

	@Override
	public int hashCode() {
		int keyCode = map.keySet().stream().map(String::hashCode).reduce(Integer::compareTo).orElse(0);
		int valueCode = map.values().stream().map(Object::hashCode).reduce(Integer::compareTo).orElse(0);
		return (keyCode + valueCode) * 31;
	}

	@Override
	public boolean equals(Object obj) {
		return obj.getClass().equals(this.getClass()) && this.hashCode() == obj.hashCode();
	}
}

File: src/org/ldccc/om3/dto/DTO.kt
package org.ldccc.om3.dto

open class DTO(open var id: Int) {
	constructor() : this(0)

//	override fun hashCode(): Int {
//		val keyCode = map.keys.stream().map(String::hashCode).reduce(Int::compareTo).orElse(0)
//		val valueCode = map.values.stream().map(Any::hashCode).reduce(Int::compareTo).orElse(0)
//		return (keyCode + valueCode) * 31
//		return 0
//	}

//	override fun equals(other: Any?): Boolean {
//		return other!!.javaClass == this.javaClass && this.hashCode() == other.hashCode()
//	}

	companion object {
		val ID: String = "ID"
	}
}
